src/main/java/com/cburch/logisim/std/ttl/Ttl74164.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Logisim-evolution - digital logic design tool and simulator
+ * Copyright by the Logisim-evolution developers
+ *
+ * https://github.com/logisim-evolution/
+ *
+ * This is free software released under GNU GPLv3 license
+ */
+
+package com.cburch.logisim.std.ttl;
+
+import com.cburch.logisim.data.BitWidth;
+import com.cburch.logisim.data.Value;
+import com.cburch.logisim.fpga.designrulecheck.netlistComponent;
+import com.cburch.logisim.instance.InstancePainter;
+import com.cburch.logisim.instance.InstancePoker;
+import com.cburch.logisim.instance.InstanceState;
+import com.cburch.logisim.instance.StdAttr;
+import com.cburch.logisim.util.GraphicsUtil;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.event.MouseEvent;
+
+public class Ttl74164 extends AbstractTtlGate {
+  /**
+   * Unique identifier of the tool, used as reference in project files.
+   * Do NOT change as it will prevent project files from loading.
+   *
+   * Identifier value must MUST be unique string among all tools.
+   */
+  public static final String _ID = "74164";
+
+  public static final int PORT_INDEX_A = 0;
+  public static final int PORT_INDEX_B = 1;
+  public static final int PORT_INDEX_QA = 2;
+  public static final int PORT_INDEX_QB = 3;
+  public static final int PORT_INDEX_QC = 4;
+  public static final int PORT_INDEX_QD = 5;
+  public static final int PORT_INDEX_CLK = 6;
+  public static final int PORT_INDEX_CLR = 7;
+  public static final int PORT_INDEX_QE = 8;
+  public static final int PORT_INDEX_QF = 9;
+  public static final int PORT_INDEX_QG = 10;
+  public static final int PORT_INDEX_QH = 11;
+
+
+
+
+  public Ttl74164() {
+    super(
+        _ID,
+        (byte) 14,
+        new byte[] {3, 4, 5, 6, 10, 11, 12, 13},
+        new String[] {
+          "A",
+          "B",
+          "QA",
+          "QB",
+          "QC",
+          "QD",
+          "Clock",
+          "Clear",
+          "QE",
+          "QF",
+          "QG",
+          "QH"
+        },
+        null);
+    super.setInstancePoker(Poker.class);
+  }
+
+  public static class Poker extends InstancePoker {
+    boolean isPressed = true;
+
+    private boolean isInside(InstanceState state, MouseEvent e) {
+      final var p = getTranslatedTtlXY(state, e);
```

Comment:
Can we have `p` replaced with more descriptive name please (multiple places)? Same for `i` below in loops